packages/api/src/services/http/http.service.ts

```diff
@@ -138,6 +151,66 @@ export class HttpService {
     return JSON_CONTENT_TYPES.some((contentType) => responseContentType.includes(contentType));
   }
 
+  /**
+   * Performs an HTTP request with the specified method and options.
+   *
+   * @param url     The URL to send the request to.
+   * @param method  The HTTP method to use (GET, POST, PUT, DELETE).
+   * @param options (Optional) An object containing the request options, including:
+   *                 - `params`: Query parameters as key-value pairs.
+   *                 - `headers`: Request headers as key-value pairs.
+   *                 - `body`: The request body.
+   * @returns A Promise that resolves with the full HttpResponse (body of type `T`, status, headers), or is rejected with an error if the request fails.
+   */
+  private requestFull<T>(
+    url: string,
+    method: 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH',
+    options: HttpOptions = {}
+  ): Promise<HttpResponse<T>> {
+    const queryString = this.buildQueryParams(options.params);
```

Comment:
Theres a lot of duplication here. Can't you extract some of the common logic with `request` method?